proposals/0000-extension-lifecycle-framework.md

```diff
@@ -0,0 +1,110 @@
+
+ * `Deprecated` extensions are considered to be design or implementation dead ends, and should not be used in new code. Deprecating an extension means that GHC intends to remove support for it in a future release (though this decision may be revised in the light of user feedback). When an extension is deprecated, the user's guide and the compiler warning _must_ include a statement about the longevity of the extension, though this need not necessarily commit to a concrete time for removal. Additionally, the user's guide and the warning should direct users to information about migrating away from the deprecated extension. Any extension will be deprecated prior to removal. Some extensions in this category might be `OverlappingInstances`, `Rank2Types`, `RecordPuns`, and `TypeInType`.
+
+ * `Legacy` extensions are explicitly not recommended for new code, much like `Deprecated`, however, they are expected to be supported indefinitely. This may be to maintain an older language set, maintain a bridge to a newer feature, or otherwise keep backward compatibility. Extensions that are `Legacy` may have some conflict with `Stable` extensions and explicitly do not need to be considered for an extension to be `Stable`. Some extensions in this category might be `CUSKs`, `DeepSubsumption`,  and `NPlusKPatterns`.
+
+Above we list expected categorizations of several extensions based on statements in the user's guide. We do not attempt to prescribe any particular categorizations as part of this proposal and simply provide a small number of expectations from the perspective of someone reading the existing documentation.
+
+The above places some restrictions on the transitions included in the extension lifecycle. The following transitions are also expected to be included:
+ * `Experimental` -> `Stable`
+ * `Experimental` -> `Deprecated`
+ * `Experimental` -> `Legacy`
+ * `Stable` -> `Legacy`
+ * `Legacy` -> `Deprecated`
+
+However, it also seems plausible that new knowledge might from time to time cause a stable extension to once again be considered experimental, e.g. in the face of soundness bugs or subtle interactions with other features. We explicitly allow for more transitions to allow for unforeseen circumstances. Restrictions are places on removal only. With this we aim to balance predictibility and flexibility. The consideration of both is important as extension classifications are first and foremost a communications channel between language designers, compiler developers, and users.
```

Comment:
" We explicitly allow for more transitions".  More than what?   Can you make this precise?

Maybe you have in mind a set of "normal" transitions, ones that are "exceptional" and ones that are "disallowed"?  I'm really not sure.

Reply:
Updated For this and the above to note the excluded transitions explicitly.

Reply:
I can't see the change here.

Reply:
The explicitly excluded transitions are listed in the block right above this paragraph now.